src/Components/Index/ServiceSelectionComponent.tsx

```diff
@@ -163,118 +154,92 @@ export class ServiceSelectionComponent extends SceneObjectBase<ServiceSelectionC
   }
 
   private updateBody() {
-    const ds = sceneGraph.lookupVariable(VAR_DATASOURCE, this)?.getValue();
     // If no services are to be queried, clear the body
     if (!this.state.servicesToQuery || this.state.servicesToQuery.length === 0) {
       this.state.body.setState({ children: [] });
     } else {
+      // If we have services to query, build the layout with the services. Children is an array of layouts for each service (1 row with 2 columns - timeseries and logs panel)
+      const children = [];
+      const favoriteServices = getFavoriteServicesFromStorage(
+        sceneGraph.lookupVariable(VAR_DATASOURCE, this)?.getValue()
+      );
+      for (const service of this.state.servicesToQuery) {
+        // for each service, we create a layout with timeseries and logs panel
+        children.push(
+          this.buildServiceLayout(service, favoriteServices.includes(service)),
+          this.buildServiceLogsLayout(service)
+        );
+      }
       this.state.body.setState({
         children: [
-          new ByLabelRepeater({
-            $data: new SceneDataTransformer({
-              $data: new SceneQueryRunner({
-                datasource: explorationDS, // explorationDS is just { uid: VAR_DATASOURCE}
-                queries: [buildVolumeQuery(this.state.servicesToQuery)],
-              }),
-              transformations: [
-                () => (source: Observable<DataFrame[]>) =>
-                  source.pipe(map((data: DataFrame[]) => orderResults(data, getFavoriteServicesFromStorage(ds)))),
-              ],
-            }),
-            body: new SceneFlexLayout({
-              height: '200px',
-              direction: 'column',
-              children: [
-                new SceneFlexItem({
-                  body: new SceneReactObject({
-                    reactNode: <LoadingPlaceholder text="Fetching services..." />,
-                  }),
-                }),
-              ],
-            }),
-            repeatByLabel: SERVICE_NAME,
-            getLayoutChild: this.getLayoutChild.bind(this),
+          new SceneCSSGridLayout({
+            children,
+            isLazy: true,
```

Comment:
🪄 magic!

Reply:
Yeah, this specifically is a huge benefit of scenes. 🪄